### PR TITLE
Dropbox plugin fix

### PR DESCRIPTION
**Pull request by xMase**
* updated dropbox api and dependency
+ added option to copy the direct link
+ added short link option

### CODE REVIEW

**Comment by olav-st** [comment]
Thanks for the pull request! I just tested it and I got the following error
```
Error in file: /home/olav/snap/screencloud/2/.local/share/screencloud/ScreenCloud/plugins/dropbox/main.py
Traceback (most recent call last):
  File "<string>", line 20, in <module>
  File "/home/olav/snap/screencloud/2/.local/share/screencloud/ScreenCloud/plugins/dropbox/modules/dropbox/__init__.py", line 3, in <module>
    from .dropbox import __version__, Dropbox, DropboxTeam, create_session  # noqa: F401
  File "/home/olav/snap/screencloud/2/.local/share/screencloud/ScreenCloud/plugins/dropbox/modules/dropbox/dropbox.py", line 17, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'
```
Looks like the requests module is missing from the Makefile. Also, the travis build is failing because of some errors with the imgur plugin. Maybe some unintended changes snuck in?

**Comment by xMase** [comment]
yes i did something wrong on merge, now i fixed it.

**Comment by olav-st** [comment]
Thanks! It appears to work now :) 

I have a suggestion for the UI. What about making the short url checkbox a radio button instead? At the moment it's possible to select both "short url" and "don't copy link", wich could be confusing for the users of the plugin. See the screenshot below:

A screenshot of the settings screen for the Dropbox plugin, with both short url and don't copy link selected.

What do you think?

File: dropbox/settings.ui [hunk 1]
```diff
@@ -105,6 +105,13 @@
       <string>Clipboard</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
+      <item>
+       <widget class="QCheckBox" name="short_url">
```
Comment (olav-st) [inline dropbox/settings.ui]:
```suggestion
       <widget class="QRadioButton" name="radio_shortlink">
```

File: dropbox/settings.ui [hunk 2]
```diff
@@ -105,6 +105,13 @@
...
+        <property name="text">
+         <string>Short Url</string>
```
Comment (olav-st) [inline dropbox/settings.ui]:
```suggestion
         <string>Copy short link (db.tt)</string>
```

File: dropbox/main.py [hunk 3]
```diff
@@ -52,6 +52,7 @@ def updateUi(self):
 			self.settingsDialog.group_name.setEnabled(True)
 			self.settingsDialog.group_clipboard.setEnabled(True)
 
+		self.settingsDialog.group_clipboard.short_url.setChecked(self.short_url)
```
Comment (olav-st) [inline dropbox/main.py]:
```suggestion
		self.settingsDialog.group_clipboard.radio_shortlink.setChecked(self.copy_shortlink)
```

File: dropbox/main.py [hunk 4]
```diff
@@ -64,6 +65,7 @@ def loadSettings(self):
 		self.access_token = settings.value("access-token", "")
 		self.user_id = settings.value("user-id", "")
 		self.display_name = settings.value("display-name", "")
+		self.short_url = settings.value("short-url", "true") in ['true', True]
```
Comment (olav-st) [inline dropbox/main.py]:
```suggestion
		self.copy_shortlink = settings.value("copy-shortlink", "true") in ['true', True]
```

File: dropbox/main.py [hunk 5]
```diff
@@ -76,8 +78,10 @@ def saveSettings(self):
 		settings.setValue("access-token", self.access_token)
 		settings.setValue("user-id", self.user_id)
 		settings.setValue("display-name", self.display_name)
+		settings.setValue("short-url", self.settingsDialog.group_clipboard.short_url.checked)
```
Comment (olav-st) [inline dropbox/main.py]:
```suggestion
		settings.setValue("copy-shortlink", self.settingsDialog.group_clipboard.radio_shortlink.checked)
```

File: dropbox/main.py [hunk 6]
```diff
@@ -102,12 +106,15 @@ def upload(self, screenshot, name):
...
 
 		f = open(tmpFilename, 'rb')
-		response = self.client.files_upload(f, '/' + ScreenCloud.formatFilename(name))
+		response = self.client.files_upload(f.read(), '/' + ScreenCloud.formatFilename(name))
 		f.close()
 		os.remove(tmpFilename)
 		if self.copy_link:
-			share = self.client.sharing_create_shared_link('/' + ScreenCloud.formatFilename(name), short_url=True)
-			ScreenCloud.setUrl(share.url)
+			share = self.client.sharing_create_shared_link('/' + ScreenCloud.formatFilename(name), short_url=self.short_url)
```
Comment (olav-st) [inline dropbox/main.py]:
```suggestion
			share = self.client.sharing_create_shared_link('/' + ScreenCloud.formatFilename(name), short_url=self.copy_shortlink)
```

File: dropbox/main.py [hunk 7]
```diff
@@ -102,12 +106,15 @@ def upload(self, screenshot, name):
...
+			if self.short_url == False: # generate direct link
```
Comment (olav-st) [inline dropbox/main.py]:
```suggestion
			if self.copy_shortlink == False: # generate direct link
```

File: dropbox/main.py [hunk 8]
```diff
@@ -52,6 +52,7 @@ def updateUi(self):
 			self.settingsDialog.group_name.setEnabled(True)
 			self.settingsDialog.group_clipboard.setEnabled(True)
 
+		self.settingsDialog.group_clipboard.short_url.setChecked(self.short_url)
 		self.settingsDialog.group_clipboard.radio_publiclink.setChecked(self.copy_link)
 		self.settingsDialog.group_clipboard.radio_dontcopy.setChecked(not self.copy_link)
```
Comment (olav-st) [inline dropbox/main.py]:
```suggestion
		self.settingsDialog.group_clipboard.radio_dontcopy.setChecked(not self.copy_link and not self.copy_shortlink)
```

**Comment by xMase** [comment]
yes good idea, approved.

**Comment by olav-st** [comment]
Thanks, merging!

**Pull request merged**